src/python/WMCore/ReqMgr/CherryPyThreads/StatusChangeTasks.py

```diff
@@ -5,10 +5,65 @@
 from __future__ import (division, print_function)
 
 from Utils.CherryPyPeriodicTask import CherryPyPeriodicTask
+from WMCore.ReqMgr.DataStructs.RequestStatus import AUTO_TRANSITION
 from WMCore.Services.WMStats.WMStatsReader import WMStatsReader
+from WMCore.Services.WorkQueue.WorkQueue import WorkQueue
 from WMCore.Services.RequestDB.RequestDBWriter import RequestDBWriter
 from WMCore.Services.WMStats.DataStruct.RequestInfoCollection import RequestInfoCollection
 
+
+def moveForwardStatus(reqDBWriter, wfStatusDict):
+    
+    for status, nextStatus in AUTO_TRANSITION.iteritems():
+        requests = reqDBWriter.getRequestByStatus([status])
+        for wf in requests:
+            stateFromGQ = wfStatusDict.get(wf, None)
+            # special case for failed workflow
+            if stateFromGQ is None:
+                continue
+            elif stateFromGQ == "failed" and status == "assigned":
```

Comment:
Actually it will be caught in the ValueError at the bottom in that case since there will be no next state, bu t i can add that for clarity.